x/metoken/keeper/redeem.go

```diff
@@ -160,7 +161,17 @@ func (k Keeper) withdrawFromLeverage(tokensToWithdraw sdk.Coin) (sdk.Coin, error
 		return sdk.Coin{}, err
 	}
 
-	uTokensToWithdraw := sdk.NewCoin(uTokensFromLeverage.Denom, availableUTokensFromLeverage)
+	if availableUTokensFromLeverage.IsZero() {
+		return coin.Zero(tokensToWithdraw.Denom), nil
+	}
+
+	uTokensToWithdraw := sdk.NewCoin(
+		uTokensFromLeverage.Denom,
+		sdk.MinInt(
+			availableUTokensFromLeverage,
+			uTokensFromLeverage.Amount,
+		),
+	)
 	if _, _, err = k.leverageKeeper.Withdraw(
```

Comment:
The first return from Withdraw is the amount of base tokens received - you may not need to use ExchangeUToken below